lib/terraform/client.rb

```diff
@@ -54,6 +54,18 @@ def output_value(list: false, name:, &block)
       ) { |value| list ? value.each(&block) : (return value) }
     end
 
+    def list_output_names
+      output_names = []
+      state_json = JSON.parse(File.read(provisioner[:state]))
```

Comment:
Is there a way to discover output names using the client? I want to avoid depending on the structure of the state file if possible.